app/components/network-site-form/component.js

```diff
@@ -1,14 +1,31 @@
 import Ember from 'ember';
 
 export default Ember.Component.extend({
-  // TODO: should children ever bubble actions through parent into route?
+  _defaultPlace() {
+    let site = this.get('site');
+
+    return {
+      formatted_address: this.get('address'),
+      geometry: {
+        location: {
+          lat() {
+            return site.get('lat');
+          },
+          lng() {
+            return site.get('lng');
+          }
+        }
+      }
+    };
+  },
+
   actions: {
     submit() {
-      const place = this.get('place');
-      const address = place.formatted_address;
-      const location = place.geometry.location;
-
-      const data = {
+      let site = this.get('site');
+      let place = this.get('place') || this._defaultPlace();
```

Comment:
Is there a better way to do this? `place` is probably always defined in the real world, but it can be `undefined` in test cases.